src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/ProviderBase/DbConnectionFactory.cs

```diff
@@ -188,5 +154,62 @@ internal bool TryGetConnection(DbConnection owningConnection, TaskCompletionSour
 
             return true;
         }
+
+        private Task<DbConnectionInternal> CreateReplaceConnectionContinuation(Task<DbConnectionInternal> task, DbConnection owningConnection, TaskCompletionSource<DbConnectionInternal> retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionPoolGroup poolGroup, CancellationTokenSource cancellationTokenSource)
+        {
+            return task.ContinueWith(
```

Comment:
Small comment @Wraith2: ContinueWith generally performs much worse than a simple async method with await (not to mention that's it's more brittle wrt exception handling etc.). I can post some benchmarking if you want.

Reply:
Though this may be totally justified if TaskContinuationOptions.LongRunning is important here - just making the general remark.

Reply:
Language supported async await would be easier to understand as well. 
Whoever wrote the original code had some reason to use imperative constructs instead of language support but I've no idea what it was.

Reply:
I vote for prioritizing 1 - it is currently a quite confusing codebase, which makes community contributions harder than the should be.

Reply:
working on it bit by bit. https://github.com/dotnet/SqlClient/pull/625 The easy part is the identical files. The complex part is the files which have opposing or complex changes.

Reply:
Sure thing, thing kind of cleanup can be done separately of course. I'd suggest that if ContinueWith continuations are being rewritten for unrelated reasons, it may be a good opportunity to just do that with async/await, but of course that's up to the team and you.

I also totally agree that merging the netfx/netcore codebases is a high-priority task.